### PR TITLE
[Feature] lake table tablets support multiple replicas

### PR DESCRIPTION
**Pull request by kevincai**
Signed-off-by: Kevin Xiaohua Cai <caixiaohua@starrocks.com>

## What type of PR is this：
- [ ] BugFix
- [X] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 6 (66.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[176]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|2|3|66.67%|[102]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 6 (83.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[176]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterJobV2Builder.java [hunk 1]
```diff
@@ -95,9 +96,10 @@ public AlterJobV2 build() throws UserException {
     }
 
     @VisibleForTesting
-    public static List<Long> createShards(int shardCount, FilePathInfo pathInfo, FileCacheInfo cacheInfo, long groupId)
+    public static List<Long> createShards(int shardCount, int replicaNum, FilePathInfo pathInfo, FileCacheInfo cacheInfo,
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterJobV2Builder.java:99]:
```suggestion
    public static List<Long> createShards(int shardCount, short replicaNum, FilePathInfo pathInfo, FileCacheInfo cacheInfo,
```

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/alter/LakeTableAlterJobV2Builder.java]: done

**Review by wyb — DISMISSED** [review]

**Review by starrocks-xupeng — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by tracymacding — DISMISSED** [review]

**Comment by tracymacding** [comment]
run starrocks_fe_unittest

**Comment by kevincai** [comment]
code rebased.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 5 / 6 (83.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[176]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|1|1|100.00%|[]|

**Review by sduzh — APPROVED** [review]

**Review by tracymacding — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by wyb — APPROVED** [review]

**Pull request merged**